core/src/main/java/org/mule/runtime/core/processor/strategy/WorkQueueProcessingStrategyFactory.java

```diff
@@ -100,6 +101,15 @@ public ReactiveProcessor onPipeline(ReactiveProcessor pipeline) {
     }
 
     @Override
+    public ReactiveProcessor onProcessor(ReactiveProcessor processor) {
+      if (processor.getProcessingType() == ASYNC) {
+        return publisher -> from(publisher).transform(processor).publishOn(fromExecutorService(ioScheduler));
```

Comment:
according to the description of `ASYNC`, shouldn't the publishing be on a cpu light rather than io?